lib/jobsub_api.py

```diff
@@ -70,7 +86,528 @@ def jobsub_call(argv: List[str], return_output: bool = False) -> Optional[str]:
+            print("", end="\r")
+
+    def find_dag_jobs(self) -> None:
+        self.dagjobs = q(
+            group=self.group, name=self.schedd, constraint=f"DAGManJobId=={self.seq}"
+        )
+
+    def fetchlog(
+        self, destdir: str = "", condor: bool = False, verbose: int = 0
+    ) -> str:
+        """fetch job output either as tarfile in current directory
+        or unpacked into directory destdir"""
+        args = ["jobsub_fetchlog"]
+        if destdir:
+            args.append("--destdir")
+            args.append(destdir)
+        if condor:
+            args.append("--condor")
+        return self._cmd(verbose, args)
+
+    def __str__(self) -> str:
+        """return jobsub_q style text line (slightly wider)"""
+        return f"{self.id:40} {self.owner:10.10} {str(self.submitted):19.19} {str(self.runtime):17} {str(self.status)[10:]:9} {self.prio:6.1f} {self.size:6.1f} {self.command}"
+
+
+jobsub_required_args = ["group"]
+
+# could we generate this from the option parser?
+jobsub_flags = {
+    "dag": "--dag",
+    "mail_always": "--mail_always",
+    "mail_on_error": "--mail_on_error",
+    "managed_token": "--managed-token",
+    "n": "-n",
+    "mail_never": "--mail-never",
+    "onsite": "--onsite",
+    "offsite": "--offsite",
+    "use_cvmfs_dropbox": "--use-cvmfs-dropbox",
+    "use_pnfs_dropbox": "--use-pnfs-dropbox",
+}
+
+# could we generate this from the option parser?
+jobsub_options = {
+    "auth_methods": "--auth-methods",
+    "blocklist": "--blocklist",
+    "c": "-c",
+    "cmtconfig": "--cmtconfig",
+    "cpu": "--cpu",
+    "dataset_definition": "--dataset-definition",
+    "dd_extra_dataset": "--dd-extra-dataset",
+    "dd_percentage": "--dd-percentage",
+    "devserver": "--devserver",
+    "disk": "--disk",
+    "d": "-d",
+    "e": "-e",
+    "email_to": "--email-to",
+    "expected_lifetime": "--expected-lifetime",
+    "f": "-f",
+    "generate_email_summary": "--generate-email-summary",
+    "group": "-G",
+    "global_pool": "--global-pool",
+    "gpu": "--gpu",
+    "i": "-i",
+    "job_info": "--job-info",
+    "L": "-L",
+    "maxConcurrent": "--maxConcurrent",
+    "memory": "--memory",
+    "need_scope": "--need-scope",
+    "need_storage_modify": "--need-storage-modify",
+    "N": "-N",
+    "no_env_cleanup": "--no-env-cleanup",
+    "OS": "--OS",
+    "overwrite_condor_requirements": "--overwrite-condor-requirements",
+    "project_name": "--project-name",
+    "resource_provides": "--resource-provides",
+    "role": "--role",
+    "r": "-r",
+    "singularity_image": "--singularity-image",
+    "site": "--site",
+    "skip_check": "--skip-check",
+    "subgroup": "--subgroup",
+    "tarball_exclusion_file": "--tarball-exclusion-file",
+    "tar_file_name": "--tar_file_name",
+    "timeout": "--timeout",
+    "t": "-t",
+    "verbose": "--verbose",
+}
+
+
+# pylint: disable=dangerous-default-value, too-many-branches, too-many-arguments, invalid-name
+def submit(
+    executable: str,
```

Comment:
Something I just realized - we require the `-G/--group` flag for `jobsub_submit`.  We should also require it here in the function signature, with a default being `os.environ.get("GROUP", None)`.

Sorry I missed this earlier.